postgresql-plugin/src/main/java/io/cdap/plugin/postgres/PostgresConnector.java

```diff
@@ -112,13 +114,25 @@ public StructuredRecord transform(LongWritable longWritable, PostgresDBRecord re
   }
 
   @Override
-  protected String getTableQuery(String database, String schema, String table) {
-    return String.format("SELECT * FROM \"%s\".\"%s\"", schema, table);
+  protected String getTableName(String database, String schema, String table) {
+    return String.format("\"%s\".\"%s\"", schema, table);
   }
 
   @Override
-  protected String getTableQuery(String database, String schema, String table, int limit) {
-    return String.format("SELECT * FROM \"%s\".\"%s\" LIMIT %d", schema, table, limit);
+  protected String getTableQuery(String database, String schema, String table, int limit, String sampleType,
+                                 String strata) {
+    if (sampleType == null) {
+      return getTableQuery(database, schema, table, limit);
+    }
+    String tableName = getTableName(database, schema, table);
+    switch (SampleType.fromString(sampleType)) {
+      case RANDOM:
+        // This query doesn't guarantee exactly "limit" number of rows
+        return String.format("SELECT * FROM %s\n" +
```

Comment:
when building queries, variables that come from user input need to be set using the PreparedStatement API (https://docs.oracle.com/javase/7/docs/api/java/sql/PreparedStatement.html) to prevent sql injection.

It's probably not practically needed here since things will likely fail earlier if somebody tries to inject sql in the table name and limit is likely validated to be a number somewhere earlier, but it is still better practice to do this using a PreparedStatement.

Reply:
Since the previous code already used Strings, and Ellen and I only have a week left, I'm thinking this change might be best in a separate PR. Do you think I could file a bug for it and leave this PR as-is?

Reply:
yes you can leave it as is. On second thought this probably isn't straightforward, as you need to pass the query to the InputFormat as a string. Don't worry about filing a bug.